src/services/authService.ts

```diff
@@ -20,32 +19,37 @@ export class AuthService implements IAuthService {
         private readonly userService: IUserService
     ){}
 
-    async signInWithGoogle(): Promise<User | undefined> {
+    async signIn(): Promise<User | undefined> {
         try {
-            const result = await signInWithPopup(this.auth, this.provider)
-            const userResponse = parseUser(result.user)
-            return await this.createUserOnFirstLogin(userResponse)
+            const credentials = await this.signInWithGoogle()
+            const existingUser = await this.checkUserExists(credentials.user.email || '')
+            if(existingUser) return existingUser
+
+            return await this.createUser(credentials.user)
         } catch (error) {
             console.error(error)
             this.alertHelper.alertError("Não foi possivel logar com o Google")
         }
     }
-    async createUserOnFirstLogin(userResponse: User): Promise<User | undefined> {
-        const findUser = await this.userService.findUser(userResponse)
 
-        if(!findUser){
-            this.userService.createUser(userResponse)
-            return(userResponse)
-        }
+    async signInWithGoogle() : Promise<UserCredential> {
```

Comment:
Hey, i had an idea to make this class even more decoupled. You have mentioned before we could have signIn with Facebook. What if we passed the provider name in signIn method, then create a factory method in the class to retrieve the provider (Google or Facebook). Finally the signin method could be renamed to signInWith Provider. 

an exemple of usage would be: signInWithProvider("Google")